app/src/components/TodoList.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <li class="list-group-item" @click="selectItem">
+    <span>{{item.title}}</span>
+    <div v-show="show" class="btn-container">
+      <button 
+        @click="removeItem(item)"
+        type="button" class="btn btn-danger"
+      >
+        <img src="../assets/trash.svg" alt="trash" width="20">
+      </button>
+      <div class="input-group mb-3 input-container">
+        <input
+          type="text"
+          class="form-control"
+          placeholder="Recipient's username"
+          v-model="item.title"
+        >
+        <div class="input-group-append">
+          <button
+            @click="updateItem(item)"
+            class="btn btn-warning"
+            type="button"
+            id="button-addon2"
+          >
+            <img src="../assets/edit.svg" alt="edit" width="20">
+          </button>
+        </div>
+      </div>
+    </div>
+  </li>
+</template>
+
+<script>
+
+export default {
+  name: "todoList",
+  props: ['item', 'show', 'index'],
+  methods: {
+    selectItem() {
+      this.$emit('select', {selectedId: this.item.id});
```

Comment:
You can send just item.id with event, no need to wrap in an object - there is only one field in this object and no possibilities to scale in future